### PR TITLE
Remove deprecated Report methods

### PR DESCRIPTION
**Pull request by rieske**
https://github.com/gradle/gradle/issues/17151

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
The updated version of SpotBugs was released over the weeked.  Running all tests one last time - the SpotBugs smoke tests was the last failure here, so this should be ready to merge.

File: subprojects/reporting/src/main/java/org/gradle/api/reporting/Report.java [hunk 1]
```diff
@@ -74,31 +71,6 @@ public String determineName(Report report) {
...
-
-    /**
-     * Whether or not this report should be generated by whatever generates it.
-     *
-     * @see #isEnabled()
-     * @param enabled Whether or not this report should be generated by whatever generates it.
-     *
-     * @deprecated Use {@link #getRequired()}.set() instead. This method will be removed in Gradle 8.0.
-     */
-    @Deprecated
```
Comment (big-guy) [inline subprojects/reporting/src/main/java/org/gradle/api/reporting/Report.java:99]: Can you also remove these from `org.gradle.api.reporting.Report.xml`?

**Review by big-guy — CHANGES_REQUESTED** [review]
Almost there

File: subprojects/reporting/src/main/java/org/gradle/api/reporting/Report.java (hunk 1 above)
Comment (tresat) [reply subprojects/reporting/src/main/java/org/gradle/api/reporting/Report.java:99]: Do you mean to remove `required` to hide it from the docs?  `Destination` and `enabled` are already gone: https://github.com/gradle/gradle/pull/17107/files#diff-69755cd09d076c67da500a7b0b78b0e27dc7fe1d8c417c6ee85686ba36878308

**Review by big-guy — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**